### PR TITLE
DEV2-1707: add a snooze button

### PR DESCRIPTION
**Pull request by yairco1990**

### CODE REVIEW

File: Common/src/main/java/com/tabnineCommon/state/CompletionsState.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+        timer?.cancel(false)
+
+        if (!isCompletionsEnabled) {
+            timer = Utils.executeThread(
+                Runnable {
+                    setCompletionsEnabled(true)
+                },
+                1,
+                TimeUnit.HOURS
```
Comment (yonip23) [inline Common/src/main/java/com/tabnineCommon/state/CompletionsState.kt:23]: is it not configurable at all currently? is it ok?

File: Common/src/main/java/com/tabnineCommon/state/CompletionsState.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package com.tabnineCommon.state
+
+import com.tabnineCommon.general.Utils
+import java.util.concurrent.Future
+import java.util.concurrent.TimeUnit
+
+object CompletionsState {
+    private var isCompletionsEnabled = true
```
Comment (yonip23) [inline Common/src/main/java/com/tabnineCommon/state/CompletionsState.kt:8]: make it atomic, as you set it from a future which is on a separate thread

File: TabnineSelfHosted/src/main/java/com/tabnineSelfHosted/statusBar/TabnineSelfHostedStatusBarWidget.kt [hunk 3]
```diff
@@ -47,6 +49,12 @@ class TabnineSelfHostedStatusBarWidget(project: Project) :
                     update()
                 }
             )
+
+        CompletionsStateNotifier.subscribe(object : CompletionsStateNotifier {
```
Comment (yonip23) [inline TabnineSelfHosted/src/main/java/com/tabnineSelfHosted/statusBar/TabnineSelfHostedStatusBarWidget.kt:53]: is this enabled only in self hosted?
just making sure you haven't forgot anything by mistake

**Review by yonip23 — APPROVED** [review]
see comments

File: Common/src/main/java/com/tabnineCommon/state/CompletionsState.kt (hunk 1 above)
Comment (yairco1990) [reply Common/src/main/java/com/tabnineCommon/state/CompletionsState.kt:23]: yes, we snooze the completions for one hour

File: Tabnine/src/main/java/com/tabnine/statusBar/TabnineStatusBarWidget.java [hunk 4]
```diff
@@ -70,14 +73,21 @@ public TabnineStatusBarWidget(@NotNull Project project) {
             }
           }
         });
+
+    CompletionsStateNotifier.Companion.subscribe(isEnabled -> update());
```
Comment (yairco1990) [inline Tabnine/src/main/java/com/tabnine/statusBar/TabnineStatusBarWidget.java:77]: @yonip23 this is where I've handled the saas

File: TabnineSelfHosted/src/main/java/com/tabnineSelfHosted/statusBar/TabnineSelfHostedStatusBarWidget.kt (hunk 3 above)
Comment (yairco1990) [reply TabnineSelfHosted/src/main/java/com/tabnineSelfHosted/statusBar/TabnineSelfHostedStatusBarWidget.kt:53]: I actually handled SAAS, look at my comment

**Review by yahave — APPROVED** [review]

File: Tabnine/src/main/java/com/tabnine/statusBar/TabnineStatusBarWidget.java (hunk 4 above)
Comment (yonip23) [reply Tabnine/src/main/java/com/tabnine/statusBar/TabnineStatusBarWidget.java:77]: oh missed it 👍🏻

**Pull request merged**